packages/core/FuzzedDataProvider.ts

```diff
@@ -372,33 +372,100 @@ export class FuzzedDataProvider {
 	 * is not sufficiently long.
 	 * @param maxLength the maximum length of the string
 	 * @param encoding the encoding of the string
+	 * @param printable - a boolean, which defaults to false that indicates whether consumed strings
+	 * should be forced to contain only valid printable characters
 	 * @returns a `string` of length between 0 and `maxLength` (inclusive)
 	 */
 	consumeString(
 		maxLength: number,
-		encoding: BufferEncoding | undefined = "ascii"
+		encoding: BufferEncoding | undefined = "ascii",
+		printable: boolean | undefined = false
 	): string {
 		if (maxLength < 0) throw new Error("maxLength must be non-negative");
+		let result;
 		const arrayLength = Math.min(maxLength, this._remainingBytes);
-		const result = this.data.toString(
-			encoding,
-			this.dataPtr,
-			this.dataPtr + arrayLength
-		);
+
+		if (printable) {
+			result = this.bufToPrintableString(
+				this.data,
+				this.dataPtr,
+				this.dataPtr + arrayLength,
+				encoding
+			);
+		} else {
+			result = this.data.toString(
+				encoding,
+				this.dataPtr,
+				this.dataPtr + arrayLength
+			);
+		}
 		this.dataPtr += arrayLength;
 		this._remainingBytes -= arrayLength;
 		return result;
 	}
 
+	/**
+	 * Wrapping subtraction function that returns a number within a set limit
+	 * @param min is a number representing the minuend
+	 * @param sub is a number representing the subtrahend
+	 * @param lim sets the upper boundary
+	 * @returns a number n with the constraint 0 <= n < lim, as long as lim >=2
+	 */
+	wrappingSub(min: number, sub: number, lim: number): number {
```

Comment:
I don't think this is needed as a stand-alone function and could be merged into `toAscii()`.
But if you really really want it, I have two further comments:
1. The number `min - sub + lim` might be negative, so the result modulo lim will also be negative.
2. The name of the function could be more descriptive. Maybe `moduloInRange` or `mapToRange`